src/CsvPayload.php

```diff
@@ -13,23 +15,102 @@ class CsvPayload extends Payload implements MiddlewareInterface
     protected $mimetype = 'text/csv';
 
     /**
-     * {@inheritdoc}
+     * The field delimiter (one character only).
+     *
+     * @var string
      */
-    protected function parse(StreamInterface $stream)
+    protected $delimiter = ",";
+
+    /**
+     * The field enclosure (one character only).
+     *
+     * @var string
+     */
+    protected $enclosure = "\"";
+
+    /**
+     * The field escape (one character only).
+     *
+     * @var string
+     */
+    protected $escape = "\\";
+
+    /**
+     * @var bool
+     */
+    private $assoc = true;
+
+    /**
+     * Configure the returned object to be converted into
+     *  - a sequential array of all CSV lines
+     *  - or a SplTempFileObject
+     *
+     * @param bool $assoc
+     *
+     * @return self
+     */
+    public function associative($assoc = true)
     {
-        if ($stream->isSeekable()) {
-            $stream->rewind();
-        }
+        $this->assoc = (bool) $assoc;
 
-        $resource = $stream->detach();
-        $data = [];
+        return $this;
+    }
 
-        while (($row = fgetcsv($resource)) !== false) {
-            $data[] = $row;
+    /**
+     * Set Csv Control characters
+     *
+     * Mimic SplFileObject::setCsvControl
+     *
+     * @see http://be2.php.net/manual/en/splfileobject.setcsvcontrol.php
+     *
+     * @param string $delimiter
+     * @param string $enclosure
+     * @param string $escape
+     *
+     * @return self
+     */
+    public function setCsvControl($delimiter = ",", $enclosure = "\"", $escape = "\\")
```

Comment:
For styling purposes, we don't use `get` names in the config methods, and it's better to split these options in three different methods. Example:

```php
$csvPayload = (new Middlewares\CsvPayload())
    ->delimiter(',')
    ->escape('\\');
```

This allows to configure the control characters independently.